File: javascript/13_loop_sum.js
var fs = require('fs');
// This solution uses the big.js library for arbitrary precision math
var big = require('../../big.js/big.js');

var f = fs.readFileSync("../data/13_loop_sum.txt");
var text = f.toString().split('\n');

var total_sum = new big(0);
text.forEach(function(line){
  total_sum = total_sum.plus(line);
})

console.log(total_sum.c.slice(0,10).join(''));

File: C++/13_loop_sum.cpp
#include <iostream>
#include <fstream>
#include <algorithm>
#include "../../ttmath-0.9.3/ttmath/ttmath.h"
#include <string>

using namespace std;

int main(){

  ttmath::Int<6> total = 0;

  ifstream file ("../data/13_loop_sum.txt");

  string line;
  while(getline(file, line)){
    ttmath::Int<6> val = line;
    total += val;
  }

  string total_string;
  total.ToString(total_string);
  cout << total_string.substr(0,10) << endl;

  return 0;
}